### PR TITLE
Enhancement/21015 review alerts fields

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#21015|

## Description 

This PR adds more information fields to vulnerability detector refactor alerts

## Logs/Alerts example

It keeps working as expected (duplicate integrity_clear alerts are not part of this issue)
image

### Example vulnerability detected alerts

```json
{"timestamp":"2023-12-29T17:15:17.269-0300","rule":{"level":10,"description":"CVE-2023-0051 affects vim","id":"23505","firedtimes":10,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"focal","ip":"192.168.33.20"},"manager":{"name":"jammy"},"id":"1703880917.2673004","decoder":{"name":"json"},"data":{"vulnerability":{"assigner":"@huntrdev","cve":"CVE-2023-0051","cvss":{"cvss3":{"base_score":"7.300000","vector":{"attack_vector":"LOCAL","availability":"HIGH","confidentiality_impact":"HIGH","integrity_impact":"HIGH","privileges_required":"LOW","scope":"UNCHANGED","user_interaction":"REQUIRED"}}},"cwe_reference":"CWE-122","enumeration":"CVE","package":{"architecture":"amd64","condition":"Package less than or equal than 0","name":"vim","source":" ","version":"2:8.1.2269-1ubuntu5.21"},"published":"2023-01-04T18:15:09Z","rationale":"Heap-based Buffer Overflow in GitHub repository vim/vim prior to 9.0.1144.","reference":"https://huntr.dev/bounties/1c8686db-baa6-42dc-ba45-aed322802de9, https://github.com/vim/vim/commit/c32949b0779106ed5710ae3bffc5053e49083ab4, http://seclists.org/fulldisclosure/2023/Mar/17, https://security.gentoo.org/glsa/202305-16, https://support.apple.com/kb/HT213670","severity":"High","status":"Active","title":"CVE-2023-0051 affects vim","type":"Packages","updated":"2023-05-03T12:16:43Z"}},"location":"vulnerability-detector"}
```

```json
{"timestamp":"2023-12-29T17:15:17.248-0300","rule":{"level":7,"description":"CVE-2022-0392 affects vim","id":"23504","firedtimes":5,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"focal","ip":"192.168.33.20"},"manager":{"name":"jammy"},"id":"1703880917.2666362","decoder":{"name":"json"},"data":{"vulnerability":{"assigner":"@huntrdev","cve":"CVE-2022-0392","cvss":{"cvss2":{"base_score":"6.800000","vector":{"access_complexity":"MEDIUM","authentication":"NONE","availability":"PARTIAL","confidentiality_impact":"PARTIAL","integrity_impact":"PARTIAL"}}},"cwe_reference":"CWE-122","enumeration":"CVE","package":{"architecture":"amd64","condition":"Package less than or equal than 0","name":"vim","source":" ","version":"2:8.1.2269-1ubuntu5.21"},"published":"2022-01-28T22:15:15Z","rationale":"Heap-based Buffer Overflow in GitHub repository vim prior to 8.2.","reference":"https://huntr.dev/bounties/d00a2acd-1935-4195-9d5b-4115ef6b3126, http://seclists.org/fulldisclosure/2022/Oct/28, http://seclists.org/fulldisclosure/2022/Oct/41, http://seclists.org/fulldisclosure/2022/Oct/43, https://lists.debian.org/debian-lts-announce/2022/11/msg00032.html, https://github.com/vim/vim/commit/806d037671e133bd28a7864248763f643967973a, https://security.gentoo.org/glsa/202208-32, https://support.apple.com/kb/HT213444, https://support.apple.com/kb/HT213488","severity":"Medium","status":"Active","title":"CVE-2022-0392 affects vim","type":"Packages","updated":"2022-11-29T22:12:29Z"}},"location":"vulnerability-detector"}
```

### Example vulnerability detected solved

```json
{"timestamp":"2023-12-29T17:14:10.382-0300","rule":{"level":3,"description":"The CVE-2023-5441 that affected vim was solved due to a package removal/update or a system upgrade","id":"23502","firedtimes":1,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"focal","ip":"192.168.33.20"},"manager":{"name":"jammy"},"id":"1703880850.2257803","decoder":{"name":"json"},"data":{"vulnerability":{"cve":"CVE-2023-5441","cvss":{"cvss3":{"base_score":"6.200000"}},"enumeration":"CVE","package":{"architecture":"amd64","name":"vim","version":"2:8.1.2269-1ubuntu5.21"},"published":"2023-10-05T21:15:11Z","reference":"https://huntr.dev/bounties/b54cbdf5-3e85-458d-bb38-9ea2c0b669f2, https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VDDWD25AZIHBAA44HQT75OWLQ5UMDKU3/, https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VGTVLUV7UCXXCZAIQIUCLG6JXAVYT3HE/, https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XPT7NMYJRLBPIALGSE24UWTY6F774GZW/, https://github.com/vim/vim/commit/20d161ace307e28690229b68584f2d84556f8960","severity":"Medium","status":"Solved","title":"CVE-2023-5441 affecting vim was solved","type":"Packages","updated":"2023-11-15T02:33:52Z"}},"location":"vulnerability-detector"}
```

```json
{"timestamp":"2023-12-29T17:14:10.598-0300","rule":{"level":3,"description":"The CVE-2022-0392 that affected vim was solved due to a package removal/update or a system upgrade","id":"23502","firedtimes":18,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"focal","ip":"192.168.33.20"},"manager":{"name":"jammy"},"id":"1703880850.2293458","decoder":{"name":"json"},"data":{"vulnerability":{"cve":"CVE-2022-0392","cvss":{"cvss2":{"base_score":"6.800000"}},"enumeration":"CVE","package":{"architecture":"amd64","name":"vim","version":"2:8.1.2269-1ubuntu5.21"},"published":"2022-01-28T22:15:15Z","reference":"https://huntr.dev/bounties/d00a2acd-1935-4195-9d5b-4115ef6b3126, http://seclists.org/fulldisclosure/2022/Oct/28, http://seclists.org/fulldisclosure/2022/Oct/41, http://seclists.org/fulldisclosure/2022/Oct/43, https://lists.debian.org/debian-lts-announce/2022/11/msg00032.html, https://github.com/vim/vim/commit/806d037671e133bd28a7864248763f643967973a, https://security.gentoo.org/glsa/202208-32, https://support.apple.com/kb/HT213444, https://support.apple.com/kb/HT213488","severity":"Medium","status":"Solved","title":"CVE-2022-0392 affecting vim was solved","type":"Packages","updated":"2022-11-29T22:12:29Z"}},"location":"vulnerability-detector"}
```

## Update 01/04/24 

image
image

```json
{"timestamp":"2024-01-04T13:57:38.710-0300","rule":{"level":3,"description":"The CVE-2020-25742 that affected qemu-utils was solved due to a package removal/update or a system upgrade","id":"23502","firedtimes":273,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"bullseye","ip":"192.168.33.90"},"manager":{"name":"jammy"},"id":"1704387458.20591471","decoder":{"name":"json"},"data":{"vulnerability":{"cve":"CVE-2020-25742","cvss":{"cvss2":{"base_score":"2.100000"}},"enumeration":"CVE","package":{"architecture":"amd64","name":"qemu-utils","version":"1:5.2+dfsg-11+deb11u3"},"published":"2020-10-06T15:15:15Z","reference":"https://bugzilla.redhat.com/show_bug.cgi?id=1883178, https://lists.nongnu.org/archive/html/qemu-devel/2020-09/msg05294.html, http://www.openwall.com/lists/oss-security/2020/09/29/1, https://ruhr-uni-bochum.sciebo.de/s/NNWP2GfwzYKeKwE?path=%2Flsi_nullptr1","severity":"Low","status":"Solved","title":"CVE-2020-25742 affecting qemu-utils was solved","type":"Packages","updated":"2020-10-07T13:32:37Z"}},"location":"vulnerability-detector"}
```

```json
{"timestamp":"2024-01-04T13:56:34.260-0300","rule":{"level":5,"description":"CVE-2020-25742 affects qemu-utils","id":"23503","firedtimes":294,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"bullseye","ip":"192.168.33.90"},"manager":{"name":"jammy"},"id":"1704387394.19975861","decoder":{"name":"json"},"data":{"vulnerability":{"assigner":"mitre","cve":"CVE-2020-25742","cvss":{"cvss2":{"base_score":"2.100000","vector":{"access_complexity":"LOW","authentication":"NONE","availability":"PARTIAL","confidentiality_impact":"NONE","integrity_impact":"NONE"}}},"cwe_reference":"CWE-476","enumeration":"CVE","package":{"architecture":"amd64","condition":"Package less than or equal than 0","name":"qemu-utils","source":"qemu","version":"1:5.2+dfsg-11+deb11u3"},"published":"2020-10-06T15:15:15Z","rationale":"pci_change_irq_level in hw/pci/pci.c in QEMU before 5.1.1 has a NULL pointer dereference because pci_get_bus() might not return a valid pointer.","reference":"https://bugzilla.redhat.com/show_bug.cgi?id=1883178, https://lists.nongnu.org/archive/html/qemu-devel/2020-09/msg05294.html, http://www.openwall.com/lists/oss-security/2020/09/29/1, https://ruhr-uni-bochum.sciebo.de/s/NNWP2GfwzYKeKwE?path=%2Flsi_nullptr1","severity":"Low","status":"Active","title":"CVE-2020-25742 affects qemu-utils","type":"Packages","updated":"2020-10-07T13:32:37Z"}},"location":"vulnerability-detector"}
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Review logs syntax and correct language
- [x] Added unit tests (for new features)

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp [hunk 1]
```diff
@@ -46,57 +46,28 @@ class UpdateCVEDescription final
...
-                                                                              : "HIGH";
-                        auto version = metricCVSSV2_0->version();
-                        VulnDescFBScoreVersionStr = (nullptr != version) ? version->str() : "";
-                        auto format = field->format();
-                        VulnDescFBClassificationStr = (nullptr != format) ? format->str() : "";
-                        break;
-                    }
-                }
-            }
+            std::string VulnDescFBAssignerStr;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp:58]: Variable names should start in lowercase

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp [hunk 2]
```diff
@@ -129,14 +100,137 @@ class UpdateCVEDescription final
...
+                    for (const auto& field : *metricsArray)
+                    {
+                        auto metricCVSSV3_1 = field->cvssV3_1();
+                        auto metricCVSSV3_0 = field->cvssV3_0();
+                        auto metricCVSSV2_0 = field->cvssV2_0();
+                        if (metricCVSSV3_1)
+                        {
+                            VulnDescFBScoreBase = metricCVSSV3_1->baseScore();
+                            auto baseSeverity = metricCVSSV3_1->baseSeverity();
+                            VulnDescFBSeverityStr = (nullptr != baseSeverity) ? baseSeverity->str() : "";
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp:114]: We can avoid the "baseSeverity" variable declaration and also the nullptr comparison to make the code cleaner.
Example of the replacement of the 2 previous lines:
`VulnDescFBSeverityStr = metricCVSSV3_1->baseSeverity()? metricCVSSV3_1->baseSeverity()->str() : "";`

These changes could be applied to all similar cases.

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp [hunk 3]
```diff
@@ -84,6 +84,13 @@ class PackageScanner final : public AbstractHandler<std::shared_ptr<ScanContext>
                                       versionStringLessThanOrEqual.c_str());
 
                             data->m_elements[callbackData.cveId()->str()] = nlohmann::json::object();
+                            data->m_alerts[callbackData.cveId()->str()] = nlohmann::json::object();
+                            if (data->getType() == ScannerType::PackageInsert)
+                            {
+                                data->m_alerts[callbackData.cveId()->str()]["vulnerability"]["package"]["condition"] =
+                                    "Package equal than " + versionString;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp:91]: Typo. Should be "equal to"

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp [hunk 4]
```diff
@@ -114,6 +121,14 @@ class PackageScanner final : public AbstractHandler<std::shared_ptr<ScanContext>
                                           versionStringLessThanOrEqual.c_str());
 
                                 data->m_elements[callbackData.cveId()->str()] = nlohmann::json::object();
+                                data->m_alerts[callbackData.cveId()->str()] = nlohmann::json::object();
+
+                                if (data->getType() == ScannerType::PackageInsert)
+                                {
+                                    data->m_alerts[callbackData.cveId()->str()]["vulnerability"]["package"]
+                                                  ["condition"] = "Package less than or equal than " + versionString;
```
Comment (matias-braida) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp:129]: Typo. Should be "equal to"

**Review by matias-braida — CHANGES_REQUESTED** [review]
GJ

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp (hunk 1 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp]: Changed

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp (hunk 2 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp]: I'm no pretty sure if it is entirely correct to treat a null as a boolean, but I did the change.

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp (hunk 3 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp]: Oops!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp (hunk 4 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp]: Oops!

**Review by matias-braida — APPROVED** [review]
LGTM

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp (hunk 2 above)
Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp]: Yes, it is valid to convert a pointer value to a boolean. Here is an extract of the C++ 11 standard:
image

**Review by GabrielEValenzuela — CHANGES_REQUESTED** [review]
After downloading this branch and executing the test behavior I found a `segfault` that breaks `modulesd`

I attach the result of gdb

```bash
gdb /var/ossec/bin/wazuh-modulesd
```

```bash
run args -ddf
2024/01/04 20:55:03 wazuh-modulesd:content-updater[68203] zipDecompressor.hpp:74 at handleRequest(): DEBUG: ZipDecompressor - Starting process
2024/01/04 20:55:03 wazuh-modulesd:content-updater[68203] zipDecompressor.hpp:49 at decompress(): DEBUG: Decompressing 'queue/vd_updater/tmp/downloads/281100_1703273628.zip' into 'queue/vd_updater/tmp/contents'
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] pubSubPublisher.hpp:61 at handleRequest(): DEBUG: PubSubPublisher - Starting process
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] pubSubPublisher.hpp:42 at publish(): DEBUG: Data to be published: '{"offset":281100,"paths":["queue/vd_updater/tmp/contents/vd_1.0.0_vd_4.8.0_281100_1703273628.json"],"stageStatus":[{"stage":"CtiSnapshotDownloader","status":"ok"},{"stage":"ZipDecompressor","status":"ok"}],"type":"raw"}'
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] pubSubPublisher.hpp:45 at publish(): INFO: Data published
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] skipStep.hpp:48 at handleRequest(): DEBUG: SkipStep - Starting process
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] cleanUpContent.hpp:63 at handleRequest(): DEBUG: CleanUpContent - Starting process
2024/01/04 20:55:14 wazuh-modulesd:vulnerability-scanner[68203] databaseFeedManager.hpp:276 at operator()(): INFO: Processing message
2024/01/04 20:55:14 wazuh-modulesd:vulnerability-scanner[68203] databaseFeedManager.hpp:141 at processMessage(): INFO: Processing file: queue/vd_updater/tmp/contents/vd_1.0.0_vd_4.8.0_281100_1703273628.json
2024/01/04 20:55:14 wazuh-modulesd:vulnerability-scanner[68203] databaseFeedManager.hpp:159 at processMessage(): DEBUG: Processing line: 1
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] CtiDownloader.hpp:204 at handleRequest(): DEBUG: CtiOffsetDownloader - Starting process
2024/01/04 20:55:14 wazuh-modulesd:content-updater[68203] CtiOffsetDownloader.hpp:42 at download(): DEBUG: Initial API offset: 281100

Thread 122 "wazuh-modulesd" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffd996c640 (LWP 68355)]
0x00007fffecbb26be in flatbuffers::ReadScalar<int> (p=0x0) at /home/administrator/Documents/wazuh/src/external/flatbuffers/include/flatbuffers/base.h:427
427	 return EndianScalar(*reinterpret_cast<const T *>(p));

(gdb) backtrace
#0  0x00007fffecbb26be in flatbuffers::ReadScalar<int> (p=0x0) at /home/administrator/Documents/wazuh/src/external/flatbuffers/include/flatbuffers/base.h:427
#1  0x00007fffecb88577 in flatbuffers::Table::GetVTable (this=0x0) at /home/administrator/Documents/wazuh/src/external/flatbuffers/include/flatbuffers/table.h:30
#2  0x00007fffecb885a8 in flatbuffers::Table::GetOptionalFieldOffset (this=0x0, field=4) at /home/administrator/Documents/wazuh/src/external/flatbuffers/include/flatbuffers/table.h:37
#3  0x00007fffecbc4d8c in flatbuffers::Table::GetPointer<flatbuffers::String const*, unsigned int> (this=0x0, field=4)
    at /home/administrator/Documents/wazuh/src/external/flatbuffers/include/flatbuffers/table.h:52
#4  0x00007fffecbb27e8 in flatbuffers::Table::GetPointer<flatbuffers::String const*, unsigned int> (this=0x0, field=4)
    at /home/administrator/Documents/wazuh/src/external/flatbuffers/include/flatbuffers/table.h:59
#5  0x00007fffecb8b55f in cve_v5::CVSS_V3_1::version (this=0x0) at /home/administrator/Documents/wazuh/src/wazuh_modules/vulnerability_scanner/include/cve5_generated.h:140
#6  0x00007fffecb9ec6a in UpdateCVEDescription::storeVulnerabilityDescription (cve5Flatbuffer=0x7fffe7b105e0, rocksDBWrapper=...)
    at /home/administrator/Documents/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVEDescription.hpp:167
#7  0x00007fffecb9ffd4 in StoreModel::handleRequest (this=0x7fffe7a16110, data=...)
    at /home/administrator/Documents/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp:47
#8  0x00007fffecbbc2a5 in AbstractHandler<std::shared_ptr<EventContext> >::handleRequest (this=0x7fffe7a142d0, data=...)
    at /home/administrator/Documents/wazuh/src/shared_modules/utils/chainOfResponsability.hpp:119
#9  0x00007fffecb99a8d in EventDecoder::handleRequest (this=0x7fffe7a142d0, data=...)
    at /home/administrator/Documents/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp:170
#10 0x00007fffecc1b9fe in TDatabaseFeedManager<IndexerConnector, PolicyManager, ContentRegister, RouterSubscriber, DatabaseFeedManagerMessageProcessor, UNIXSocketRequest>::TDatabaseFeedManager(std::shared_ptr<IndexerConnector>, std::atomic<bool> const&, bool)::{lambda(std::vector<char, std::allocator<char> > const&)#1}::operator()(std::vector<char, std::allocator<char> > const&) const::{lambda(nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long, unsigned long, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > const&)#1}::operator()(nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long, unsigned long, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > const&) const (__closure=0x7fffe7a25c80, resource=...)
    at /home/administrator/Documents/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:260
#11 0x00007fffecc24f43 in std::__invoke_impl<void, TDatabaseFeedManager<IndexerConnector, PolicyManager, ContentRegister, RouterSubscriber, DatabaseFeedManagerMessageProcessor, UNIXSocketRequest>::TDatabaseFeedManager(std::shared_ptr<IndexerConnector>, std::atomic<bool> const&, bool)::{lambda(std::vector<char, std::allocator<char> > const&)#1}::operator()(std::vector<char, std::allocator<char> > const&) const::{lambda(nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long, unsigned long, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > const&)#1}&, nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long, unsigned long, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > const&>(std::__invoke_other, TDatabaseFeedManager<IndexerConnector, PolicyManager, ContentRegister, RouterSubscriber, DatabaseFeedManagerMessageProcessor, UNIXSocketRequest>::TDatabaseFeedManager(std::shared_ptr<IndexerConnector>, std::atomic<bool> const&, bool)::{lambda(std::vector<char, std::allocator<char> > const&)#1}::operator()(std::vector<char, std::allocator<char> > const&) const::{lambda(nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long, unsigned long, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > const&)#1}&, nlohmann::json_abi_v3_11_2::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long, unsigned long, double, std::allocator, nlohmann::json_abi_v3_11_2::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > co
```

In particular, the line appears to be `updateCVEDescription.hpp:167`


```cpp
vulnDescFBScoreVersionStr =
                                (metricCVSSV3_1->version()) ? metricCVSSV3_1->version()->str() : "";
```

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM! The overall structure is clear, and the code is easy to read and understand. The `segfault` was fixed successfully!

Great job Migue!

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp [hunk 5]
```diff
@@ -114,6 +117,9 @@ class PackageScanner final : public AbstractHandler<std::shared_ptr<ScanContext>
                                           versionStringLessThanOrEqual.c_str());
 
                                 data->m_elements[callbackData.cveId()->str()] = nlohmann::json::object();
+                                data->m_alerts[callbackData.cveId()->str()]["vulnerability"]["package"]["condition"] =
+                                    "Package less than or equal to " + versionString;
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp:121]: You need to use versionStringLessThan or versionLessThanOrEqual

Ex

```suggestion
                                    versionStringLessThan.empty() ? "Package less than or equal to "  + versionStringLessThanOrEqual : "Package less than " + versionStringLessThan;
```

**Review by Dwordcito — CHANGES_REQUESTED** [review]
image

**Pull request merged**